eng/pipelines/templates/steps/retain-heap-dump-hprofs.yml

```diff
@@ -0,0 +1,17 @@
+steps:
+  - pwsh: |
+      New-Item $(Build.ArtifactStagingDirectory)/oom-hprofs -ItemType directory
+      foreach($hprofFile in (Get-ChildItem -Path . -Recurse -Filter *.hprof -File))
+      {
+        $fileFullName = $hprofFile.FullName
+        $fileName = $hprofFile.Name
+        Copy-Item $fileFullName $(Build.ArtifactStagingDirectory)/oom-hprofs/$fileName -ErrorAction SilentlyContinue
+      }
+      [System.IO.Compression.ZipFile]::CreateFromDirectory("$(Build.ArtifactStagingDirectory)/oom-hprofs","$(Build.ArtifactStagingDirectory)/oom-hprofs.zip")
```

Comment:
In general creating a zip file make uploading less efficient as Devops already does compression on the server side and can optimize the chunks of these files better across all files uploaded instead of us.

Reply:
I'd like to see how that plays out for real OOM dumps.   The test dumps were highly compressible  2gb -> 8mb I think.  If they're not compressing on upload, then the build's waiting on a 2gb file transfer.

Reply:
I'm less concerned on the DevOps slow down for uploading and more on the downloading when the dev loop includes troubleshooting an OOM. Overall, OOMs should rarely happen and I want to focus on the developer experience when they do.

Reply:
From a download prospective Devops always has the artifacts in a compressed download format so I'd expect them to be downloaded in a zip file regardless of how you upload them. In fact if you upload a zip you will likely end up with a zip inside of a zip.

Reply:
However with that said I'm fine testing out either approach and we can make adjustments as necessary.